File: hw/chavez/u3/hw22SOLIDMaze/hw22Java/MazeSystem/src/main/java/ec/edu/espe/mazesystem/controller/MazeController.java
package ec.edu.espe.mazesystem.controller;

import ec.edu.espe.mazesystem.model.AbstractMaze;
import ec.edu.espe.mazesystem.model.Maze;
import ec.edu.espe.mazesystem.renderer.MazeRenderer;
import ec.edu.espe.mazesystem.generator.MazeGenerator;

public class MazeController {
    private AbstractMaze model;
    private MazeRenderer renderer;
    private int currentWidth;
    private int currentHeight;

    public MazeController(MazeRenderer renderer) {
        this.renderer = renderer;
        this.currentWidth = 5;
        this.currentHeight = 5;
        this.model = new Maze(currentWidth, currentHeight, new MazeGenerator());
    }

    public MazeController(AbstractMaze model, MazeRenderer renderer) {
        this.model = model;
        this.renderer = renderer;
        if (model != null) {
            this.currentWidth = model.getWidth();
            this.currentHeight = model.getHeight();
        }
    }

    public void setRenderer(MazeRenderer renderer) {
        this.renderer = renderer;
    }

    public void initMazeGeneration(int width, int height) {
        this.currentWidth = width;
        this.currentHeight = height;
        
        Maze maze = new Maze(width, height, new MazeGenerator());
        maze.generate();
        this.model = maze;
        
        if (renderer != null) {
            renderer.render(maze);
        }
    }

    public void renderMaze() {
        if (renderer != null && model != null) {
            renderer.render(model);
        }
    }

    public AbstractMaze getModel() {
        return model;
    }

    public MazeRenderer getRenderer() {
        return renderer;
    }

    public void setModel(AbstractMaze model) {
        this.model = model;
        if (model != null) {
            this.currentWidth = model.getWidth();
            this.currentHeight = model.getHeight();
        }
    }

    public int getCurrentWidth() {
        return currentWidth;
    }

    public int getCurrentHeight() {
        return currentHeight;
    }
}

File: hw/tipantiza/u3/hw22MazeSolidPrinciples/MazeSystem_py/controller/maze_controller.py
from model.maze import Maze
from generator.maze_generator import MazeGenerator


class MazeController:

    def __init__(self, renderer, model=None):
        self.renderer = renderer

        if model is None:
            self.current_width = 5
            self.current_height = 5
            self.model = Maze(
                self.current_width,
                self.current_height,
                MazeGenerator()
            )
        else:
            self.model = model
            self.current_width = model.get_width()
            self.current_height = model.get_height()

    def set_renderer(self, renderer):
        self.renderer = renderer

    def init_maze_generation(self, width, height):
        self.current_width = width
        self.current_height = height

        maze = Maze(
            width,
            height,
            MazeGenerator()
        )

        maze.generate()

        self.model = maze

        if self.renderer is not None:
            self.renderer.render(maze)

    def render_maze(self):
        if self.renderer is not None and self.model is not None:
            self.renderer.render(self.model)

    def get_model(self):
        return self.model

    def get_renderer(self):
        return self.renderer

    def set_model(self, model):
        self.model = model

        if model is not None:
            self.current_width = model.get_width()
            self.current_height = model.get_height()

    def get_current_width(self):
        return self.current_width

    def get_current_height(self):
        return self.current_height